pkg/repository/ensurer.go

```diff
@@ -108,8 +107,11 @@ func (r *Ensurer) repoLock(key BackupRepositoryKey) *sync.Mutex {
 
 func (r *Ensurer) createBackupRepositoryAndWait(ctx context.Context, namespace string, backupRepoKey BackupRepositoryKey) (*velerov1api.BackupRepository, error) {
 	toCreate := NewBackupRepository(namespace, backupRepoKey)
-	if err := veleroclient.CreateRetryGenerateName(r.repoClient, ctx, toCreate); err != nil {
-		return nil, errors.Wrap(err, "unable to create backup repository resource")
+
+	if err := r.repoClient.Create(ctx, toCreate, &client.CreateOptions{}); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
```

Comment:
you don't need the nested if here.
`if err != nil && !apierrors.IsAlreadyExists(err)`

Reply:
Modified